src/api/api-service.ts

```diff
@@ -46,4 +46,8 @@ export class ApiService {
       this.headers,
     );
   }
+
+  public async validateUsername(username: string) {
+    return await HttpClient.get(`${this.apiEndpoint}/util/${username}`);
```

Comment:
this endpoint takes a post method
```suggestion
    return await HttpClient.post(`${this.apiEndpoint}/util/${username}`);
```